src/presentations/Header.js

```diff
@@ -0,0 +1,124 @@
+            state.itemCount= props.cartInfo.length;
+        }
+        if(props.productInfo && props.productInfo.cartItems){
+            state.itemCount= props.productInfo.cartItems.length
+        }
+        return state;
+    }
+    resize=()=>{
+        this.doResizeActions();
+    }
+    resetCart=()=>{
+        this.props.resetCartData();
+    }
+    doResizeActions=()=>{
+        let dimensions = getViewPortDimensions();
+        this.setState({dimensions: dimensions});
+    }
+    cartClick=()=>{
+        if(this.state.dimensions && this.state.dimensions.width && this.state.dimensions.width < 480){ //1024
+            this.props.router.push('/cart');
+        } else {
+            this.props.cartClick();
+        }
+    }
+    clickEvent=(event)=>{}
+    showHome=()=>{
+        this.props.router.push('/');
+    }
+    render(){   
+        const {userName,itemCount} = this.state;
+        return (<div className="header-area">
+            <header>
+                <nav className="sticky-nav">
+                    <div className="row">
+                        <img onClick={this.showHome} src={window.location.origin + '/static/images/logo.png'} alt="logo" className="logo" /> {/* {Logo} */}
+                        <ul className="main-nav js--main-nav">
+                            <li><Link to={`/home`} >Home</Link></li>
+                            <li><Link to="/products">Products</Link></li>
+                            <div className="logo-icon-header">
+                            <li><Link to={`/home`} ><i className="ion-ios-home"></i></Link></li>
+                            <li><Link to="/products"><i className="ion-ios-list"></i></Link></li>
+                            <li><Link to={`/signin`} ><i className="ion-ios-person"></i></Link></li>
+                            <li><Link to="/signup"><i className="ion-ios-personadd"></i></Link></li>
+                            </div>
+                           
+                        </ul>
+                        <div className="guest-info">
+                            <div>
+                            {userName && userName.length ? (<div> 
+                                <p className="useremail" title={userName} aria-label={userName}>{userName}</p>
+                            </div>) : (<ul className="main-nav js--main-nav">
+                                <li onClick={this.resetCart}><Link to="/signin">Signin</Link></li>
+                                <li onClick={this.resetCart}><Link to="/signup">Register</Link></li>
+                            </ul>)}
+                            </div>
+                            <div className="cartArea" aria-labelledby="cart" onClick={this.cartClick}>
+                                <div className="cart-icon">
+                                    <LazyLoad height={25} once>
+                                        <img src={window.location.origin + '/static/images/cart.svg'} className="fill-purple"></img>
```

Comment:
add alt

Reply:
fix string literal creation

Reply:
An alt attribute is given to cart img tag. If we give a relative path to an image it's not rendering images. So I was told to give with window.location.origin with the pathname where an image is located to work. So, I kept this way.